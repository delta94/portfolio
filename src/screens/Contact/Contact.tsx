import clsx from 'clsx';
import React from 'react';
import { createUseStyles } from 'react-jss';
import { useMediaQuery } from 'react-responsive';
import 'react-responsive-modal/styles.css';
import { Element } from 'react-scroll';
import { FormContainer, MapContainer, VanishAnimation } from '../../components';
import { CustomTheme } from '../../theme';

interface ContactProps {}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  header: {
    color: theme.colors.white,
    marginLeft: theme.spacing(3),
    display: 'flex',
    justifyContent: 'center',
    cursor: 'pointer',
  },
  formContainer: {
    display: 'flex',
    justifyContent: 'center',
    flex: 1,
  },
  mapContainer: {
    display: 'flex',
    width: '50%',
    justifyContent: 'center',
    marginLeft: theme.spacing(3),
  },
  mapContainerMobile: {
    width: '90%',
    marginLeft: theme.spacing(0),
    alignSelf: 'center',
  },
  contactUsContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
  contactUsContainerMobile: {
    flexDirection: 'column',
  },
}));

const Contact: React.FC<ContactProps> = () => {
  const classes: Record<
    | 'header'
    | 'contactUsContainer'
    | 'contactUsContainerMobile'
    | 'formContainer'
    | 'mapContainer'
    | 'mapContainerMobile',
    string
  > = useStyles();

  const isMobile: boolean = useMediaQuery({ maxWidth: 1118 });

  return (
    <Element name='Contact'>
      <div className={classes.header}>
        <VanishAnimation words={['Contact']} />
      </div>
      <div className={clsx([classes.contactUsContainer, isMobile && classes.contactUsContainerMobile])}>
        <div className={clsx([classes.mapContainer, isMobile && classes.mapContainerMobile])}>
          <MapContainer />
        </div>
        <div className={classes.formContainer}>
          <FormContainer />
        </div>
      </div>
    </Element>
  );
};

export { Contact };
