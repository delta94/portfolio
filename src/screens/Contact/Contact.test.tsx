import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { Contact } from './Contact';

test('Contact renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <Contact />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
