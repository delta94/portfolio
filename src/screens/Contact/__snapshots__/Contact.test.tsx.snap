// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `
<DocumentFragment>
  <div
    name="Contact"
  >
    <div
      class="header-0-2-1"
    >
      <div
        style="opacity: 0; transform: translate3d(0,20px,0);"
      >
        <h2
          class="h2Text-0-2-15"
        >
          Contact
        </h2>
      </div>
    </div>
    <div
      class="formContainer-0-2-2"
    >
      <form
        class="form-0-2-3"
      >
        <div
          class="smallInputFieldContainer-0-2-5"
        >
          <div
            class="smallInputFieldParentContainer-0-2-6"
          >
            <input
              class="smallInputField-0-2-7"
              name="name"
              placeholder="Your Name"
              type="text"
            />
          </div>
          <div
            class="smallInputFieldParentContainer-0-2-6"
          >
            <input
              class="smallInputField-0-2-7"
              name="email"
              placeholder="Email address"
              type="email"
            />
          </div>
        </div>
        <div
          class="biggerInputFieldContainer-0-2-8"
        >
          <input
            class="smallInputField-0-2-7"
            name="subject"
            placeholder="Subject"
            type="text"
          />
        </div>
        <textarea
          class="smallInputField-0-2-7 textArea-0-2-9"
          name="description"
          placeholder="Type your message here."
        />
        <div
          class="buttonContainer-0-2-10"
        >
          <button
            class="mainContainer-0-2-16"
            style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
          >
            Send Message
          </button>
          <div
            class="emailTextContainer-0-2-11"
          >
            Don't link forms? Send me an
            <span
              class="email-0-2-12"
            >
              Email
            </span>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
